src/event-manager/index.ts

```diff
@@ -0,0 +1,79 @@
+import { EventEmitter } from 'events';
+import Ain from '../ain';
+import {
+  BlockchainEventTypes,
+  BlockFinalizedEventConfig,
+  EventChannelConnectionOption,
+  EventConfigType,
+  ValueChangedEventConfig,
+} from '../types';
+import EventFilter from './event-filter';
+import EventChannelClient from './event-channel-client';
+import { PushId } from '../ain-db/push-id';
+import EventCallbackManager from './event-callback-manager';
+
+export default class EventManager {
+  private _ain: Ain;
+  private readonly _filters: { [filterId: string]: EventFilter };
+  private readonly _eventEmitters: { [filterId: string]: EventEmitter };
+  private readonly _eventCallbackManager: EventCallbackManager;
+  private readonly _eventChannelClient: EventChannelClient;
+
+  constructor(ain: Ain) {
+    this._filters = {};
+    this._eventEmitters = {};
+    this._ain = ain;
+    this._eventCallbackManager = new EventCallbackManager(this);
+    this._eventChannelClient = new EventChannelClient(ain, this._eventCallbackManager);
+  }
+
+  async connect(connectionOption?: EventChannelConnectionOption) {
+    await this._eventChannelClient.connect(connectionOption || {});
+  }
+
+   getEventEmitter(filterId: string) {
+    return this._eventEmitters[filterId];
+  }
+
+  disconnect() {
+    this._eventChannelClient.disconnect();
+  }
+
+  buildFilterId() {
+    return PushId.generate();
+  }
+
+  subscribe(
+      eventType: 'BLOCK_FINALIZED',
+      config: BlockFinalizedEventConfig): EventEmitter;
+  subscribe(
+      eventType: 'VALUE_CHANGED',
+      config: ValueChangedEventConfig): EventEmitter;
+  subscribe(eventTypeStr: string, config: EventConfigType): EventEmitter {
+    if (!this._eventChannelClient.isConnected) {
+      throw Error(`Event channel is not connected! You must call ain.eh.connect() before using subscribe()`);
+    }
+    const eventType = eventTypeStr as BlockchainEventTypes;
+    const eventEmitter = new EventEmitter();
+    switch (eventType) {
```

Comment:
what about moving the the parts related with this._filters and this._eventEmitters to event callback manager as it's the role of the event callback manager by definition?

Reply:
Hmm... Also, how about moving `this._filters` and `this._eventEmitters` to event callback manager?

Reply:
yes, yes